### DISCUSSION TITLE
Evaluate expression in scope

### DISCUSSION THREAD

**Discussion by m7mdtiger** [josdejong/mathjs#2866] [Ideas]
Hello, 

I try to use evaluate() but it through an error if the value in scope is an expression

```
let scope = {
  x: "y + 4",
  y: "c+ 4",
  c: "2"
}
console.log(mathjs.evaluate(scope.x, scope)) # Through an Error: Cannot convert "c + 4" to a number
```

I think that it would be nice to support solving such cases.. 
```
let resolvedScope = {};

while(Object.keys(scope).length > 0) {
    for(const key in scope) {
        try {
          resolvedScope[key] = mathjs.evaluate(scope[key], resolvedScope);
          delete scope[key];
        } catch(error) {
          // continue
        }
    }
}
```
Thank you.

**Comment by josdejong** [comment]
The expression parser does not evaluate variables from the scope as an expression.

It would be interesting though to think though implementing a solver that can resolve these kind of Excel-like expressions.

See also: 
- https://github.com/josdejong/mathjs/issues/904
- https://github.com/josdejong/mathjs/discussions/2689

**Reply by m7mdtiger** [reply]
Thank you.
